include/zephyr/arch/xtensa/cache.h

```diff
@@ -127,11 +128,11 @@ static ALWAYS_INLINE uint32_t z_xtrpoflip(uint32_t addr, uint32_t rto, uint32_t
  * @param ptr A pointer to a valid C object
  * @return A pointer to the same object via the L1 dcache
  */
-static inline void *arch_xtensa_cached_ptr(void *ptr)
+static inline void __cache *arch_xtensa_cached_ptr(void *ptr)
 {
-	return (void *)z_xtrpoflip((uint32_t) ptr,
-				   CONFIG_XTENSA_CACHED_REGION,
-				   CONFIG_XTENSA_UNCACHED_REGION);
+	return (__force void __cache *)z_xtrpoflip((uint32_t) ptr,
```

Comment:
This is very intrusive, especially when our default static analysis tool is not "sparse" ...

It can also be very confusing; i.e. "force" and "cache" can mean many different things -- for example, one might think `__cache` is a Zephyr attribute that places a variable or function inside a cached memory region, which this certainly does not.

While I understand that `__force` is from Linux, "sparse" is their default static analysis tool while it is not for Zephyr.

In that sense, IMO, we should prefix these macros such that it is clear for the users that they are for the "sparse" tool (e.g. `__sparse_force`, `__sparse_cache`).

Reply:
@stephanosio while I don't mind in principle making those macros sparse-specific, that's just very long... Any chance for anything shorter? Especially for `__cache` since SOF will have quite a few of them: https://github.com/thesofproject/sof/pull/5359

Reply:
yeah, that saves 3 characters but makes it much less readable IMHO. I went with the original suggestion, please, review